spec/support/test_helpers/teacher_training_public_api_helper.rb

```diff
@@ -24,6 +24,16 @@ def stub_teacher_training_api_providers(recruitment_cycle_year: RecruitmentCycle
     )
   end
 
+  def stub_teacher_training_api_providers_for_pagination(recruitment_cycle_year: RecruitmentCycle.current_year)
+    api_response1 = pagination_response(1)
+    api_response2 = pagination_response(2)
+    api_response3 = pagination_response(3)
+
+    stub_pagination_request(recruitment_cycle_year, 1, api_response1)
+    stub_pagination_request(recruitment_cycle_year, 2, api_response2)
+    stub_pagination_request(recruitment_cycle_year, 3, api_response3)
```

Comment:
```suggestion
page_numbers = [1,2,3]

page_numbers.each do |page_number|
  stub_pagination_request(recruitment_cycle_year, page_number, pagination_response(page_number))
end
```

Reply:
This would be a more simplified way of writing this but i think slightly less clear. Thoughts? perhaps I could change the method name to be clearer...

Reply:
I prefer this suggested version and agree that the method names could help.

`stub_teacher_training_api_providers_with_multiple_pages`?